markdown/developers/adding_modules.md

```diff
@@ -291,7 +299,18 @@ The key words "MUST", "MUST NOT", "SHOULD", etc. are to be interpreted as descri
 
 ### General
 
-- All non-mandatory command-line tool options MUST be provided as a string i.e. `options.args` where `options` is a Groovy Map that MUST be provided via the Nextflow `addParams` option when including the module via `include` in the parent workflow.
+- All non-mandatory command-line tool options MUST be provided as a string via the `$args` variable, where `args` is a Groovy Map that MUST be provided via the Nextflow `task.ext.args` variable.
+
+   ```nextflow
+   script:
+      def args = task.ext.args ?: ''
+      def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
+   """
+   fastqc \\
+     $args \\
+     <...>
+   """
+   ```
```

Comment:
```suggestion
    `<module>.nf`:
    ```nextflow
    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
    """
    fastqc \\
        $args \\
         <...>
    """
    ```
    `modules.config`:
    ```nextflow
    process {
        withName: <module> {
            ext.args = [                                                                                                // Assign either a string, closure which returns a string
                '--quiet',
                params.fastqc_kmer_size ? "-k ${params.fastqc_kmer_size}" : ''     // parameter dependent values can be provided like so
            ].join(' ')                                                                                                    // join converts the list here to a string.
            ext.suffix = { "${meta.id}" }                                                     // A closure can be used to access variables defined in the script
        }
    }
    ```
```